measurement/signals/handlers.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth import get_user_model
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from guardian.shortcuts import assign_perm
+
+from measurement.models import MeasurementBase
+
+User = get_user_model()
+
+
+@receiver(post_save, sender=MeasurementBase)
+def set_permissions(sender, instance, **kwargs):
+    """Set object-level permissions"."""
+    # Get permissions for model
+    delete, change, view = _get_perm_codenames(sender)
+
+    # Assign change and delete permissions for owner
+    for perm in [change, delete]:
+        assign_perm(perm, instance.owner, instance)
```

Comment:
`instance` is an instance of the model sending the `post_save` signal, `Measurement` or `Report`, which do not have an owner. I believe you would need to use `instance.station.owner`.